scheduler.go

```diff
@@ -91,15 +91,17 @@ func (s *Scheduler) scheduleNextRun(job *Job) {
 	defer job.Unlock()
 	now := s.time.Now(s.loc)
 
-	if job.startsImmediately {
-		job.nextRun = now
-		job.startsImmediately = false
-		return
+	if job.neverRan() {
+		if !job.nextRun.IsZero() {
+			return // scheduled for future run and should skip scheduling
+		}
+		// default is for jobs to start immediately unless scheduled at a specific time or day
+		if job.atTime == 0 && job.scheduledWeekday == nil && job.dayOfTheMonth == 0 {
```

Comment:
Nit: Probably best to put this in a function. Just like we have `job.neverRan`.
maybe `job.shouldStartImmediately()`

Reply:
Or perhaps you can retain `job.startsImmediately` and make it `true` by default (when we construct the job with `New()`)